### PR TITLE
[kmac] Add shared secret key

### PR DESCRIPTION
**Pull request by eunchan**
Revise the register description to have two share secret key in the
Memory mapped registers. It means, the software shall prepare masked
secret key and write into `KEY_SHARE0` and `KEY_SHARE1`.

This PR has two commits. Second commit is only for generating the register RTL.

### CODE REVIEW

**Review by udinator — APPROVED** [review]
LGTM

File: hw/ip/kmac/data/kmac.hjson [hunk 1]
```diff
@@ -251,12 +251,12 @@
...
+        name: "KEY_SHARE0"
         desc: '''KMAC Secret Key
 
               KMAC secret key can be up to 512 bit.
               Order of the secret key is:
-              key[512:0] = {KEY0, KEY1, KEY2, ... , KEY15};
+              key[512:0] = {KEY15, KEY14, ... , KEY0};
 
               The registers are allowed to be updated when the engine is in Idle state.
               If the engine computes the hash, it discards any attempts to update the secret keys
```
Comment (alphan) [inline hw/ip/kmac/data/kmac.hjson:262]: Would defining a regwen bit be simpler?

**Review by alphan — COMMENTED** [review]
Thanks @eunchan! LGTM, just a few questions.

File: hw/ip/kmac/data/kmac.hjson (hunk 1 above)
Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:262]: I've also considered that... The history was:
1. Key to be stored as masked form, so the key written to the KMAC needs to go through masking logic (at this time assumed key stored in FW unmasked)
2. After discussed with SW ppl, looks like masked registers are better, so just added one more masked register `KEY_SHARE1` here. and... KMAC core logic already has secret key implemented, so just kept `hwext` as true :)

File: hw/ip/kmac/data/kmac.hjson [hunk 2]
```diff
@@ -277,20 +277,72 @@
...
+              KMAC secret key can be up to 512 bit.
+              Order of the secret key is:
+              key[512:0] = {KEY15, KEY14, ... , KEY0};
+
+              The registers are allowed to be updated when the engine is in Idle state.
+              If the engine computes the hash, it discards any attempts to update the secret keys
+              and report an error.
+
+              Current KMAC supports up to 512 bit secret key. It is the sw
+              responsibility to keep upper bits of the secret key to 0.
```
[the remark this answers is not in the document]
Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:295]: Hum, good questions. At this time, I implemented in that ways described above. But your point looks good. I will revise the logic and this description only to use valid bits.

File: hw/ip/kmac/data/kmac.hjson [hunk 3]
```diff
@@ -277,20 +277,72 @@
...
             '''
+      regwen: "CFG_REGWEN"
       swaccess: "wo"
       hwaccess: "hro"
       hwext: "false"
       fields: [
-        { bits: "9:0"
+        { bits: "2:0"
           name: "len"
           desc: "Length in bits"
```
[the remark this answers is not in the document]
Comment (eunchan) [reply hw/ip/kmac/data/kmac.hjson:323]: Thanks, I will revise.

**Pull request merged**